Marlin/temperature.cpp

```diff
@@ -1949,5 +1956,7 @@ void Temperature::isr() {
     }
   #endif
 
+  cli();
```

Comment:
cli() is not clearing all interrupts, it clears only the bit that controls interrupts are executed. So it's disabling them, but if an interrupt event is triggered it will be executed after the next sei(). It will also not stop the already running interrupt handler code.
The reason why I added it is the following. After the cli, we say the temperature ISR is finished by setting in_temp_isr = false and we re-enable the temp ISR itself. So the temperature ISR is not protected against re-entry in this areay anymore. By executing cli() before, I ensure there will be no further ISR execution until this ISR realy has finished and the code returns to its last position. (sei is done during the return from each ISR)